server/mergin/sync/storages/disk.py

```diff
@@ -214,56 +242,61 @@ def apply_diff(
         basefile = os.path.join(self.project_dir, current_file.location)
         changeset = os.path.join(self.project_dir, upload_file.diff.location)
         patchedfile = os.path.join(self.project_dir, upload_file.location)
-        # create copy of basefile which will be updated in next version
+        # create local copy of basefile which will be updated in next version and changeset needed
         # TODO this can potentially fail for large files
         logging.info(f"Apply changes: copying {basefile} to {patchedfile}")
         start = time.time()
-        copy_file(basefile, patchedfile)
-        copy_time = time.time() - start
-        logging.info(f"Copying finished in {copy_time} s")
-        try:
-            # clean geodiff logger
-            self.flush_geodiff_logger()
-            logging.info(
-                f"Geodiff: apply changeset {changeset} of size {os.path.getsize(changeset)} with changes to {patchedfile}"
-            )
-            start = time.time()
-            self.geodiff.apply_changeset(patchedfile, changeset)
-            geodiff_apply_time = time.time() - start
-            # track performance of geodiff action
-            base_version = current_file.location.split("/")[0]
-            gh = GeodiffActionHistory(
-                self.project.id,
-                base_version,
-                current_file.path,
-                current_file.size,
-                v_name,
-                "apply_changes",
-                changeset,
-            )
-            gh.copy_time = copy_time
-            gh.geodiff_time = geodiff_apply_time
-            logging.info(f"Changeset applied in {geodiff_apply_time} s")
-        except (GeoDiffLibError, GeoDiffLibConflictError):
-            move_to_tmp(patchedfile)
-            move_to_tmp(changeset)
-            return Err(self.gediff_log.getvalue())
-
-        # we can now replace old basefile metadata with the new one (patchedfile)
-        # TODO this can potentially fail for large files
-        logging.info(f"Apply changes: calculating checksum of {patchedfile}")
-        start = time.time()
-        checksum = generate_checksum(patchedfile)
-        checksumming_time = time.time() - start
-        gh.checksum_time = checksumming_time
-        logging.info(f"Checksum calculated in {checksumming_time} s")
-        db.session.add(gh)
-        return Ok(
-            (
-                checksum,
-                os.path.getsize(patchedfile),
-            )
-        )
+        with self.geodiff_copy(changeset) as changeset_tmp, self.geodiff_copy(
+            basefile
+        ) as patchedfile_tmp:
```

Comment:
this is basefile_tmp no?

Reply:
no, it is not - we copy basefile as a new patched file because geodiff modifies this file with changeset